contracts/cross-chain/IZNSZChainPortal.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.26;
+
+import { IPolygonZkEVMBridgeV2 } from "@zero-tech/zkevm-contracts/contracts/v2/interfaces/IPolygonZkEVMBridgeV2.sol";
+import { IDistributionConfig } from "../types/IDistributionConfig.sol";
+import { IZNSRootRegistrarTrunk } from "../registrar/IZNSRootRegistrarTrunk.sol";
+import { IZNSSubRegistrarTrunk } from "../registrar/IZNSSubRegistrarTrunk.sol";
+import { IZNSRegistry } from "../registry/IZNSRegistry.sol";
+import { IZNSTreasury } from "../treasury/IZNSTreasury.sol";
+import { IZNSChainResolver } from "../resolver/IZNSChainResolver.sol";
+import { IZNSAccessController } from "../access/IZNSAccessController.sol";
+
+
+interface IZNSZChainPortal is IDistributionConfig {
+
+    struct ZNSContractInput {
+        IZNSAccessController accessController;
+        IZNSRegistry registry;
+        IZNSChainResolver chainResolver;
+        IZNSTreasury treasury;
+        IZNSRootRegistrarTrunk rootRegistrar;
+        IZNSSubRegistrarTrunk subRegistrar;
+    }
+
+    event DomainBridged(
+        uint32 indexed destNetworkId,
+        address destPortalAddress,
+        bytes32 indexed domainHash,
+        address indexed domainOwner
+    );
+    event DestZnsPortalSet(address newAddress);
+
+    error DestinationPortalNotSetInState();
+
+    function polygonZkEVMBridge() external view returns (IPolygonZkEVMBridgeV2);
+
+    function destNetworkId() external view returns (uint32);
+
+    function destChainName() external view returns (string memory);
+
+    function destChainId() external view returns (uint256);
+
+    function destZnsPortal() external view returns (address);
+
+    function rootRegistrar() external view returns (IZNSRootRegistrarTrunk);
```

Comment:
Consider moving this and a few of the other items that are shared between this contract and `IZNSEthereumPortal` to a higher level `IZNSPortal`, to avoid code duplication